code/player/controller/DefaultWalkController.cs

```diff
@@ -43,9 +52,75 @@ public override void Simulate()
                 SprintSpeed = (MaxSprintSpeed - DefaultSpeed) / player.MaxStamina * player.Stamina + DefaultSpeed;
             }
 
+            IsDiving = Pawn.WaterLevel.Fraction == 1f;
+
+            if (!IsDiving)
+            {
+                TimeSinceDivingStarted = 0f;
+            }
```

Comment:
Do we want to do these checks on server? It might be better to see if the player is underwater on server as opposed to client.

Reply:
Controller should be predicted in any way. So stuff that is needed to calculate predicted stuff has to be the same on both sides.  I could make this var predicted, but as it's clientside just used for smooth visual effects and serverside used to adjust the health / dealing damage, I think this is the best way avoiding unnecessary network traffic.